pkg/trace/obfuscate/json.go

```diff
@@ -87,15 +108,29 @@ func (p *jsonObfuscator) obfuscate(data []byte) (string, error) {
 		case scanObjectValue, scanArrayValue:
 			// done scanning value
 			p.setKey()
-			if p.keeping && depth < p.keepDepth {
+			if p.transformingValue && p.transformer != nil {
+				v, err := strconv.Unquote(string(valBuf))
+				if err != nil {
+					v = string(valBuf)
+				}
+				result := p.transformer(v)
+				out.Write([]byte(`"`))
```

Comment:
Can we use `WriteByte` and `WriteString` here?

Reply:
Ah didn't know those existed. Updated